src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2041,44 +2041,83 @@ private BoundExpression BindArgumentValue(DiagnosticBag diagnostics, ArgumentSyn
             }
 
             var declarationExpression = (DeclarationExpressionSyntax)argumentSyntax.Expression;
-            var declaration = (TypedVariableComponentSyntax)declarationExpression.VariableComponent;
-            var typeSyntax = declaration.Type;
+            var typeSyntax = declarationExpression.Type();
+            bool isVar;
+
+            // Is this a local?
+            SourceLocalSymbol localSymbol = this.LookupLocal(declarationExpression.Identifier());
 
-            if (InConstructorInitializer || InFieldInitializer)
+            if ((object)localSymbol != null)
             {
-                Error(diagnostics, ErrorCode.ERR_ExpressionVariableInConstructorOrFieldInitializer, declarationExpression);
-            }
+                if (InConstructorInitializer || InFieldInitializer)
```

Comment:
I think the right reason for this to work is that you actually can have a { } scope around a variable initialization in scripting.

---
In reply to: [79077182](https://github.com/dotnet/roslyn/pull/13818#discussion_r79077182) [](ancestors = 79077182)